tests/UnifiedSpecTests/UnifiedSpecTest.php

```diff
@@ -252,17 +264,35 @@ private function checkRunOnRequirements(array $runOnRequirements)
 
         $serverVersion = $this->getCachedServerVersion();
         $topology = $this->getCachedTopology();
+        $serverParameters = $this->getCachedServerParameters();
 
         foreach ($runOnRequirements as $o) {
             $runOnRequirement = new RunOnRequirement($o);
-            if ($runOnRequirement->isSatisfied($serverVersion, $topology)) {
+            if ($runOnRequirement->isSatisfied($serverVersion, $topology, $serverParameters)) {
                 return;
             }
         }
 
+        // @todo Add server parameter requirements?
         $this->markTestSkipped(sprintf('Server version "%s" and topology "%s" do not meet test requirements', $serverVersion, $topology));
     }
 
+    /**
+     * Return the server parameters (cached for subsequent calls).
+     */
+    private function getCachedServerParameters()
+    {
+        static $cachedServerParameters;
+
+        if (isset($cachedServerParameters)) {
+            return $cachedServerParameters;
+        }
+
+        $cachedServerParameters = $this->getServerParameters();
```

Comment:
If unified spec tests are the only thing that access server parameters, would it make sense to just move the logic from FunctionalTestCase into this function? We can always split it out later if the need arises. This will be more helpful when I need to rebase #806, as the UnifiedTestRunner class no longer inherits FunctionalTestCase.

Reply:
Moved.